Sources/StreamChat/Controllers/MessageController/MessageController.swift

```diff
@@ -278,8 +278,11 @@ public extension ChatMessageController {
             skipPush: skipPush,
             extraData: extraData
         ) { result in
+            if let newMessage = try? result.get() {
+                self.client.eventNotificationCenter.process(NewMessagePendingEvent(message: newMessage))
+            }
```

Comment:
Yes, a comment makes sense 👍  How do you think it feels hacky? Our Event system can easily support events we create ourselves, not just web socket events. 

This one is needed because we need to communicate to the ChannelVC that a user has pressed "send" to a message so that we can act on the fact that we need to load the first page in case the user sent a message while in a mid-page. 

Maybe we can move this event to another place 🤔 Maybe MessageSender is a better candidate? It would actually be better since it works for both the Channel and the Thread.

Keep in mind that the ChannelController instance, which is used for the Composer is different from the one used in the Channel View. So there's no way to directly communicate to the Channel View, that the composer created a new message. Ofc, we could add a ComposerDelegate, but that seems like overkill. And since we have a `MessageNewEvent` already when the message is published, it makes sense to have one when the message was created but not yet published.

Reply:
The approach on using Events to communicate within the SDK reminds me a lot of the bad old pattern of sending NSNotifications. The problem with it is that it is not a direct communication pattern, and that brings us to the second issue which is that you cannot guarantee whether there is anyone listening/handling those, nor that only one party is doing so. Therefore the reach of those events cannot be controlled.

I understand that this is due to the current structure we have, such as having multiple controllers per "screen", which we already discussed in the past, and I believe we should be addressing, but it is important to also understand if this is the direction we want for the SDK moving forward.

For me, it is fine if we ship like this, but acknowledging that this is not the right pattern for our SDK to follow, and creating the necessary tickets to address this in the future.

Reply:
I can also add a new delegate method to the `ChannelControllerDelegate` to keep the EventsController only for websocket related stuff. WDYT? @polqf

Reply:
We already have this ticket somewhere, I think it makes sense to link this PR and explain in more detail how it would solve this issue. Even if we used one controller in both the Composer and the Channel View, this would not work since our ChannelController can only have one delegate. So right now, this is the only way to notify the Channel View that a message was sent to the server.